framework/base/ArrayableTrait.php

```diff
@@ -121,6 +122,8 @@ public function extraFields()
      */
     public function toArray(array $fields = [], array $expand = [], $recursive = true)
     {
+        $serializer = new Serializer();
```

Comment:
Duplication. Maybe it's better to put it in the method?

```php
private function createSerializer() {
   $serializer = new Serializer();
   $serializer->preserveKeys = true;
   return $serializer;
}

```

Reply:
I think this isn't going to be merged. Objects serializable should just be treated as non-serializable
See this https://github.com/yiisoft/yii2/issues/17889#issuecomment-591518449